src/main/java/BankingSystem.java

```diff
@@ -0,0 +1,163 @@
+                case "2" -> logIntoAccount();
+                default -> System.out.println("You've entered invalid menu item.\n");
+            }
+        }
+    }
+
+    private void welcomeMenu() {
+        System.out.println("""
+                1. Create an account
+                2. Log into account
+                0. Exit""");
+    }
+
+    private void registerAccount() {
+        Card card = cardGenerator.generateCredentials();
+        dao.save(new Account(card, 0));
+        System.out.printf("""
+                Your card has been created
+                Your card number:
+                %s
+                Your card PIN:
+                %s""", card.number(), card.pin());
+
+        System.out.println();
+        System.out.println();
+    }
+
+    private void logIntoAccount() {
+        System.out.println("\nEnter your card number:");
+        String inputtedCardNumber = scanner.next();
+        System.out.println("Enter your PIN:");
+        String inputtedCardPIN = scanner.next();
+        boolean isValid = accountValidation(inputtedCardNumber, inputtedCardPIN);
+
+        if (isValid) {
+            System.out.println("\nYou have successfully logged in!\n");
+            dao.get(inputtedCardNumber, inputtedCardPIN).ifPresent(account -> currentAccount = account);
+            accountMenu();
+        } else {
+            System.out.println("\nWrong card number or PIN!\n");
+        }
+    }
+
+    private boolean accountValidation(String inputtedCardNumber, String inputtedCardPIN) {
+        return dao.get(inputtedCardNumber, inputtedCardPIN).isPresent();
+    }
+
+    private void accountMenu() {
+        while (currentAccount != null) {
+            System.out.println("""
+                    1. Balance
+                    2. Add income
+                    3. Do transfer
+                    4. Close account
+                    5. Log out
+                    0. Exit""");
+
+            String input = scanner.next();
+            System.out.println();
+
+            switch (input) {
+                case "0" -> {
+                    dao.close();
+                    System.out.println("Bye!");
+                    System.exit(0);
+                }
+                case "1" -> getBalance();
+                case "2" -> addIncome();
+                case "3" -> doTransfer();
+                case "4" -> closeAccount();
+                case "5" -> logOut();
+                default -> System.out.println("You've entered invalid menu item.\n");
+            }
+        }
+    }
+
+    private void getBalance() {
+        System.out.printf("Balance: %d\n", currentAccount.balance());
+    }
+
+    private void addIncome() {
+        System.out.println("Enter income:");
+        int income = Integer.parseInt(scanner.next());
+        dao.update(currentAccount.card().number(), income);
+        dao.get(currentAccount.card().number(), currentAccount.card().pin()).ifPresent(account -> currentAccount = account);
```

Comment:
Can you move it to a separate method? Looks like the same block is used in line 149.

Reply:
Looks like you only need to update `balance` in the `currentAccount`.

Reply:
Made `Account` a simple POJO with balance setter instead of record